server/routes.go

```diff
@@ -1092,12 +1092,12 @@ func streamResponse(c *gin.Context, ch chan any) {
 }
 
 // ChatPrompt builds up a prompt from a series of messages for the currently `loaded` model
-func chatPrompt(ctx context.Context, messages []api.Message) (string, error) {
+func chatPrompt(ctx context.Context, model *Model, messages []api.Message) (string, error) {
 	encode := func(s string) ([]int, error) {
 		return loaded.runner.Encode(ctx, s)
 	}
 
-	prompt, err := ChatPrompt(loaded.Model.Template, loaded.Model.System, messages, loaded.Options.NumCtx, encode)
+	prompt, err := ChatPrompt(model.Template, model.System, messages, loaded.Options.NumCtx, encode)
```

Comment:
Shouldn't we be updating the `loaded` Model instead? Otherwise more issues will come from having out of date `loaded.Model` state.

Reply:
Actually, this is the only place `loaded.Model` is used, so why don't we take out that field